### PR TITLE
fix(system-backup): unexpected bulk deletion

### PR DESCRIPTION
**Pull request by c3y1huang**
https://github.com/longhorn/longhorn/issues/6045

### CODE REVIEW

File: s3/s3.go [hunk 1]
```diff
@@ -105,13 +105,22 @@ func (s *BackupStoreDriver) GetURL() string {
 }
 
 func (s *BackupStoreDriver) updatePath(path string) string {
-	return filepath.Join(s.path, path)
+	joinedPath := filepath.Join(s.path, path)
+
+	if !strings.HasSuffix(path, "/") {
```
Comment (innobead) [inline s3/s3.go:110]: Suggest adding a comment to explain why do this way.

**Review by innobead — CHANGES_REQUESTED** [review]
LGTM, but need a comment.

**Review by innobead — APPROVED** [review]

**Comment by innobead** [comment]
@c3y1huang work with QA to check if this issue also happened to other backup stores.

**Comment by c3y1huang** [comment]
@mergify backport v1.4.x

**Pull request merged**